katello-attach-subscription

```diff
@@ -171,259 +800,1046 @@ def vdcupdate()
     auto_attach = false
     keep_virt_only = false
     virtual_host = nil
+    desired_quantity = 1
+    sys_socket = 1
+    hypervisor_found = false
+    system_type = ""
+    this_system_cluster = ""
+    skip_hypervisor = false
+    sys = nil
+    skip_report = false
+    sub_counted = []
+
+    # add to array system the name of the system itself taken from id
+    if @options[:verbose]
+      puts " VERBOSE: Current system #{currentcount+1}/#{systemstotal}: #{system['name']} (#{system['id']})"
+    end
 
-    # get detail for each system
+    # get detail for the current system to be checked.
     if @options[:debug]
-      puts "DEBUG: detail of the current system to be checked:"
+      puts " DEBUG: detail of the current system to be checked:"
       p system
     end
-    sys = api.resource(@host_resource).call(:show, {:id => system['id'], :fields => 'full'})
-    # add to array system the name of the system itself taken from id
-    puts "Current system: #{system['name']} (#{system['id']})"
-
+    sys = @api.resource(:hosts).call(:show, {:id => system['id'].to_i, :fields => 'full'})
+    if sys.count <= 0
+      puts "  WARNING: System name '#{system['name']}' not found. Is cache in use? Skipping."
+      next
+    end
+    # if host isn't registered by subscription manager for anyone reason, skip
+    if not sys.has_key?('subscription_facet_attributes')
+      if @options[:verbose]
+        puts "  VERBOSE: Skipping #{sys['name']} as it isn't registered with subscription manager"
+      end
+      next
+    end
+    if @options[:debug]
+      puts "  DEBUG: '#{system['name']}' id #{system['id']} :hosts api data:"
+      p sys
+    end
+    system_details[system['name']]=sys
+    # check if the type requested match the host one
+    system_type = typefromsatellitedata(sys)
     # for each item in yaml extract sub
     if @options[:debug]
-      puts "DEBUG: YAML dump"
+      puts " DEBUG: YAML dump with all definitions for the current system"
       p @yaml
     end
     @yaml[:subs].each do |sub|
+      # if "type" has been specified on yaml file check if match.
+      # the default is to check on "Hypervisor"
+      # if DO NOT match, skip to next sub
+      # fixme: can't find this detail in new API
+      if sub.has_key?('type')
+        desired_type = sub['type']
+      end
+      # extract current information from the system
+      if system_type != desired_type
```

Comment:
desired_type defaults to `nil` if it's not been set by the sub. but this does not check for `nil`, and thus I end up with all hosts skipped:

```
  VERBOSE: Skipping 'upgrade64.katello.lan' as system type 'Guest' is different from desired ''
```